source/MRMesh/MROpenVDBHelper.h

```diff
@@ -0,0 +1,205 @@
+#pragma once
+#include "MRPch/MROpenvdb.h"
+#include "openvdb/tree/TreeIterator.h"
+#include "openvdb/tree/Tree.h"
+#include "openvdb/tree/ValueAccessor.h"
+
+namespace MR
+{
+
+template <class TreeT, typename Transformer>
+class RangeProcessor
+{
+public:
+    using InterruptFunc = std::function<bool( void )>;
+
+    using ValueT = typename TreeT::ValueType;
+
+    using LeafIterT = typename TreeT::LeafCIter;
+    using TileIterT = typename TreeT::ValueAllCIter;
+    using LeafRange = typename openvdb::tree::IteratorRange<LeafIterT>;
+    using TileRange = typename openvdb::tree::IteratorRange<TileIterT>;
+
+    using InTreeAccessor = typename openvdb::tree::ValueAccessor<const TreeT>;
+    using OutTreeAccessor = typename openvdb::tree::ValueAccessor<TreeT>;
+
+    RangeProcessor( const openvdb::math::CoordBBox& b, const TreeT& inT, TreeT& outT, const Transformer& xform ) :
+        mIsRoot( true ), mXform( xform ), mBBox( b ),
+        mInTree( inT ), mOutTree( &outT ), mInAcc( mInTree ), mOutAcc( *mOutTree )
+    {}
+
+    RangeProcessor( const openvdb::math::CoordBBox& b, const TreeT& inTree, const Transformer& xform ) :
+        mIsRoot( false ), mXform( xform ), mBBox( b ),
+        mInTree( inTree ), mOutTree( new TreeT( inTree.background() ) ),
```

Comment:
mb use std::unique_ptr and don't use `new` (alse clear destructor)

Reply:
it's not work without some inconvenient rework